pkg/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -0,0 +1,173 @@
+	return d.OsClient.ClusterRoleBindings().Get(name)
+}
+
+func (d *AggregatedLogging) routes(project string, options kapi.ListOptions) (*routesapi.RouteList, error) {
+	return d.OsClient.Routes(project).List(options)
+}
+
+func (d *AggregatedLogging) serviceAccounts(project string, options kapi.ListOptions) (*kapi.ServiceAccountList, error) {
+	return d.KubeClient.ServiceAccounts(project).List(options)
+}
+
+func (d *AggregatedLogging) services(project string, options kapi.ListOptions) (*kapi.ServiceList, error) {
+	return d.KubeClient.Services(project).List(options)
+}
+
+func (d *AggregatedLogging) endpointsForService(project string, service string) (*kapi.Endpoints, error) {
+	return d.KubeClient.Endpoints(project).Get(service)
+}
+
+func (d *AggregatedLogging) daemonsets(project string, options kapi.ListOptions) (*kapisext.DaemonSetList, error) {
+	return d.KubeClient.DaemonSets(project).List(kapi.ListOptions{LabelSelector: loggingInfraFluentdSelector.AsSelector()})
+}
+
+func (d *AggregatedLogging) nodes(options kapi.ListOptions) (*kapi.NodeList, error) {
+	return d.KubeClient.Nodes().List(kapi.ListOptions{})
+}
+
+func (d *AggregatedLogging) pods(project string, options kapi.ListOptions) (*kapi.PodList, error) {
+	return d.KubeClient.Pods(project).List(options)
+}
+func (d *AggregatedLogging) deploymentconfigs(project string, options kapi.ListOptions) (*deployapi.DeploymentConfigList, error) {
+	return d.OsClient.DeploymentConfigs(project).List(options)
+}
+
+func (d *AggregatedLogging) Info(id string, message string) {
+	d.result.Info(id, message)
+}
+
+func (d *AggregatedLogging) Error(id string, err error, message string) {
+	d.result.Error(id, err, message)
+}
+
+func (d *AggregatedLogging) Debug(id string, message string) {
+	d.result.Debug(id, message)
+}
+
+func (d *AggregatedLogging) Warn(id string, err error, message string) {
+	d.result.Warn(id, err, message)
+}
+
+func (d *AggregatedLogging) Name() string {
+	return AggregatedLoggingName
+}
+
+func (d *AggregatedLogging) Description() string {
+	return "Check aggregated logging integration for proper configuration"
+}
+
+func (d *AggregatedLogging) CanRun() (bool, error) {
+	if len(d.MasterConfigFile) == 0 {
+		return false, errors.New("No master config file was provided")
+	}
+	if d.OsClient == nil {
+		return false, errors.New("config must include a cluster-admin context to run this diagnostic")
+	}
+	if d.KubeClient == nil {
+		return false, errors.New("config must include a cluster-admin context to run this diagnostic")
+	}
+	var err error
+	d.masterConfig, err = hostdiag.GetMasterConfig(d.result, d.MasterConfigFile)
+	if err != nil {
+		return false, errors.New("Unreadable master config; skipping this diagnostic.")
+	}
+	return true, nil
+}
+
+func (d *AggregatedLogging) Check() types.DiagnosticResult {
+	project := retrieveLoggingProject(d.result, d.masterConfig, d.OsClient)
+	if len(project) != 0 {
+		checkServiceAccounts(d, d, project)
```

Comment:
@sosiouxme changed it to facilitate testing of the code by introducing an interface 'diagnosticReporter' that would allow verifying various messages were produced.  It may be there is a way to utilize the actual code and retrieve the results from where it saves them.  I sort of envision moving this reporter to a higher, public interface so it could be used across the spectrum of diagnostics for testing.